### PR TITLE
Support running on PostgreSQL 15

### LINKED ISSUE #648: Run doctests and update tests on PostgreSQL 15
Currently TimescaleDB doesn't support PostgreSQL 15, which our update tests and doctests rely on. When a TimescaleDB release with PostgreSQL 15 support comes out, we should reenable these tests in CI.

see: #646

### PR DESCRIPTION
**Pull request by Smittyvb**
Fixes #648.

- [x] Fix all doctests that call `setseed`
- [x] Fix TimescaleDB crash in CI
- [x] Fix `random` function on pg12/pg13
- [x] Don't try to run upgrade tester on releases without PG15 binaries (currently this is all releases)

Things to fix in release-build-scripts (these don't block this PR):
- Drop Ubuntu 18.04 support (since TimescaleDB has also dropped support) (https://github.com/timescale/release-build-scripts/pull/94)
- Fix rockylinux 9 error in build scripts (`package curl-minimal-7.76.1-19.el9.x86_64 conflicts with curl provided by curl-7.76.1-19.el9.x86_64`)

This reverts commit ff66df629eb0f8a0ee33dfd80e6df107b67c9e73.

### CODE REVIEW

**Comment by Smittyvb** [comment]
The [CI for this PR fails](https://github.com/timescale/timescaledb-toolkit/actions/runs/3732933453/jobs/6333040403) on PG15:

```
running 15 test files
Starting stateless tests
Starting docs/asap.md
Starting docs/stats_agg.md
Finished docs/asap.md
error dropping DB connection closed
Starting docs/client.md
Finished docs/stats_agg.md
thread '<unnamed>' panicked at 'could not connect to postgres: Error { kind: Db, cause: Some(DbError { severity: "FATAL", parsed_severity: Some(Fatal), code: SqlState(E57P03), message: "the database system is in recovery mode", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("postmaster.c"), line: Some(2403), routine: Some("ProcessStartupPacket") }) }', tools/sql-doctester/src/runner.rs:79:18
```

**Comment by Smittyvb** [comment]
The CI passes when run with the updated Docker image: https://github.com/timescale/timescaledb-toolkit/actions/runs/3830536476

File: docker/ci/setup.sh [hunk 1]
```diff
@@ -131,13 +131,10 @@ EOF
                 yum -q -y install \
```
Comment (epgts) [inline docker/ci/setup.sh:131]: This file looks good,but can you move everything else into a separate pull request?

File: tools/dependencies.sh [hunk 2]
```diff
@@ -8,8 +8,6 @@
 # information across all those.
 
 PG_VERSIONS='12 13 14 15'
-# TODO: remove this once TimescaleDB supports PostgreSQL 15: issue #648
-TSDB_PG_VERSIONS='12 13 14'
```
Comment (epgts) [inline tools/dependencies.sh:12]: Let's hold off removing this until all references are gone in this repository and in release-scripts.

File: docs/percentile_approximation.md [hunk 3]
```diff
@@ -22,23 +22,26 @@ CREATE TABLE response_times (
 );
 -- and we'll make it a hypertable for ease of use in the rest of the example
 SELECT create_hypertable('response_times', 'ts');
+-- utilities for generating random numbers
+CREATE SEQUENCE rand START 567;
+CREATE FUNCTION test_random() RETURNS float AS
+    'SELECT ((nextval(''rand'')*34567)%1000)::float/1000'
+LANGUAGE SQL;
```
Comment (epgts) [inline docs/percentile_approximation.md:29]: Can we have doctester define this for us?

File: docker/ci/setup.sh (hunk 1 above)
Comment (Smittyvb) [reply docker/ci/setup.sh]: I split off the changes to this file to https://github.com/timescale/timescaledb-toolkit/pull/670.

File: .github/workflows/patch_build.yml [hunk 4]
```diff
@@ -46,7 +46,7 @@ jobs:
       - name: Run Doc Tests
         run: |
           docker run -d --name toolkit_test -e POSTGRES_HOST_AUTH_METHOD=trust -p 5432:5432 timescaledev/timescale-analytics:nightly
-          cargo run --manifest-path ./tools/sql-doctester/Cargo.toml -- -h localhost -s "CREATE EXTENSION timescaledb_toolkit; SET SESSION TIMEZONE TO 'UTC'" -p 5432 docs
+          cargo run --manifest-path ./tools/sql-doctester/Cargo.toml -- -h localhost -p 5432 docs
```
Comment (Smittyvb) [inline .github/workflows/patch_build.yml:49]: This CI job has bitrotted and always fails to run (it's only run when certain files are modified). The patch Docker image should probably be completely removed in a future PR.

File: docs/percentile_approximation.md (hunk 3 above)
Comment (Smittyvb) [reply docs/percentile_approximation.md:29]: I updated doctester to have the startup script be a part of doctester itself.,

File: .github/workflows/patch_build.yml (hunk 4 above)
Comment (rtwalker) [reply .github/workflows/patch_build.yml:49]: motion seconded

Comment (epgts) [reply .github/workflows/patch_build.yml:49]: I asked @JLockerman if I could get rid of it the first time I tripped over it (PR #442).  At the time, he thought it was still useful, though I can't find any written record of our conversation about it.

I think it's pretty clear we don't need it now...

**Review by epgts — APPROVED** [review]

**Review by WireBaron — APPROVED** [review]

File: .github/workflows/patch_build.yml (hunk 4 above)
Comment (JLockerman) [reply .github/workflows/patch_build.yml:49]: Yeah this thing only ever had a very specific use–getting a version of the current TimescaleDB docker image with the latest dev version of Toolkit installed–and was only ever useful as an internal debugging tool. It was always rather brittle, and if we're not finding it useful anymore we might as well delete it.
